src/NUnitFramework/framework/Internal/Builders/DefaultTestCaseBuilder.cs

```diff
@@ -141,7 +145,7 @@ public Test BuildFrom(IMethodInfo method, Test parentSuite)
                     tests.Add(test);
             }
 
-            return tests.Count > 0
+            return shouldParameterizeForFewerThanTwoTests || tests.Count > 0
```

Comment:
@CharliePoole continuing our conversation, this commit will cause `TheoryWithNoDatapointsIsNotRunnable` and `UnsupportedNullableTypeArgumentWithNoDatapointsAreNotRunnable` to fail with:

>   Expected: <NUnit.Framework.Internal.TestMethod>
>   But was:  <NUnit.Framework.Internal.ParameterizedMethodSuite>

I assume this isn't just overzealous testing in these two tests?

These are all the things that `builders` may contain:
 * **`TheoryAttribute`**
 * `SequentialAttribute`
 * `PairwiseAttribute`
 * `CombinatorialAttribute`
 * `TestCaseSourceAttribute`
 * `TestCaseAttribute`
 * Future implementations of `ITestBuilder`

Alternatively, we could introduce `ITestBuilder.FailIfNoCases` and check `builders.Any(_ => _.FailIfNoCases)`. `TheoryAttribute` would be the only current test builder to return true for `FailIfNoCases`.